core/src/main/scala/org/http4s/headers/Content-Length.scala

```diff
@@ -12,18 +12,24 @@ import org.http4s.util.Writer
   *
   * @param length the length; throws an `IllegalArgumentException` if negative
   */
-final case class `Content-Length`(length: Long) extends Header.Parsed {
-  require(length >= 0L)
-
+sealed abstract case class `Content-Length`(length: Long) extends Header.Parsed {
   override def key: `Content-Length`.type = `Content-Length`
   override def renderValue(writer: Writer): writer.type = writer.append(length)
+  def modify(f: Long => Long): `Content-Length` = new `Content-Length`(f(length)) {}
```

Comment:
Now that I think about this, this a bogus method. It would let you build a negative content length